### ISSUE TITLE
fszmq NuGet package should specify version for FSharp.Core dependency

### ISSUE THREAD

**Issue by wjrogers** [zeromq/fszmq#152]
After converting a C# project from `packages.config` to the new `project.json` NuGet reference format, building a project that references fszmq 12.2.2 produces output like this:
```
Primary reference "FSharp.Core, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a".
    Resolved file path is "C:\Users\Will\.nuget\packages\FSharp.Core\2.0.0\lib\FSharp.Core.dll".
    Reference found at search path location "{RawFileName}".
    Found related file "C:\Users\Will\.nuget\packages\FSharp.Core\2.0.0\lib\FSharp.Core.xml".
    This reference is not "CopyLocal" because at least one source item had "Private" set to "false" and no source items had "Private" set to "true".
    The ImageRuntimeVersion for this reference is "v2.0.50727".
Primary reference "fszmq, Version=12.2.2.0, Culture=neutral, PublicKeyToken=null".
    Resolved file path is "C:\Users\Will\.nuget\packages\fszmq\12.2.2\lib\net40\fszmq.dll".
    Reference found at search path location "{RawFileName}".
    Found related file "C:\Users\Will\.nuget\packages\fszmq\12.2.2\lib\net40\fszmq.xml".
    This reference is not "CopyLocal" because at least one source item had "Private" set to "false" and no source items had "Private" set to "true".
    The ImageRuntimeVersion for this reference is "v4.0.30319".
There was a conflict between "FSharp.Core, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" and "FSharp.Core, Version=4.3.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a".
    "FSharp.Core, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" was chosen because it was primary and "FSharp.Core, Version=4.3.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" was not.
    References which depend on "FSharp.Core, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" [C:\Users\Will\.nuget\packages\FSharp.Core\2.0.0\lib\FSharp.Core.dll].
        C:\Users\Will\.nuget\packages\FSharp.Core\2.0.0\lib\FSharp.Core.dll
          Project file item includes which caused reference "C:\Users\Will\.nuget\packages\FSharp.Core\2.0.0\lib\FSharp.Core.dll".
            C:\Users\Will\.nuget\packages\FSharp.Core\2.0.0\lib\FSharp.Core.dll
    References which depend on "FSharp.Core, Version=4.3.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" [C:\Program Files (x86)\Reference Assemblies\Microsoft\FSharp\.NETFramework\v4.0\4.3.1.0\FSharp.Core.dll].
        C:\Users\Will\.nuget\packages\fszmq\12.2.2\lib\net40\fszmq.dll
          Project file item includes which caused reference "C:\Users\Will\.nuget\packages\fszmq\12.2.2\lib\net40\fszmq.dll".
            C:\Users\Will\.nuget\packages\fszmq\12.2.2\lib\net40\fszmq.dll
```
Because the fszmq package doesn't specify a version for its dependency, NuGet chooses the _lowest available version_, which is 2.0.0. Even though the fszmq.dll assembly itself references a newer version of the FSharp.Core.dll assembly (4.3.1.0), the package reference wins because it's considered a "primary reference," whereas the assembly reference is a simple binary dependency. FSharp.Core.dll version 2.0.0.0 gets copied to the build output folder.

To get the correct version, I have to add an explicit reference to the FSharp.Core package in my project even though I don't directly depend on it.

Reference for `project.json` and NuGet dependency resolution:
https://docs.microsoft.com/en-us/nuget/schema/project-json
https://docs.microsoft.com/en-us/nuget/consume-packages/dependency-resolution

**Comment by pblasucci** [comment]
@wjrogers thanks for reporting this. I'll see about getting explicit dependency version info into the `.nuspec`. But it might take a bit as the whole thing is currently being handled by Paket. 

Also, is this blocking you? In other words, do I need to do a point-release just for this, or can the fix wait until I've notched up a few changed to the library as well?

**Comment by wjrogers** [comment]
No, this isn't blocking me because there is a straightforward work-around (referencing FSharp.Core myself). Thanks!

**Comment by pblasucci** [comment]
Fixed in PR #153 (updated release coming soon).
